### DISCUSSION TITLE
Special Type for Addresses

### DISCUSSION THREAD

**Discussion by fayalalebrun** [SpinalHDL/SpinalHDL#1432] [Ideas]
With my current design, a tedious step and possible source of error is conversion between addresses with different granularities/word sizes. For example, some addresses are granular to 1 byte, 4 bytes, 16 bytes, etc.

I think it may be quite helpful to have a type which keeps track of this granularity and allows users to easily convert between them. It could be a simple wrapper for UInt.

I am wondering if others have encountered this issue and would find such an addition useful.

**Comment by Dolu1990** [comment]
> I am wondering if others have encountered this issue

Yes. Overall, over time i started to use byte based address for memory busses as much as i can.

> I think it may be quite helpful to have a type which keeps track of this granularity and allows users to easily convert between them. It could be a simple wrapper for UInt.

I agree that could work well.
